kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java

```diff
@@ -121,8 +129,8 @@ public String getActiveProcessImage(String containerId, long procInstId, String
         if (imageSVG != null) {
             // find active nodes and modify image
             Map<String, String> subProcessLinks = new HashMap<>();
-            Collection<NodeInstanceDesc> activeLogs = dataService.getProcessInstanceHistoryActive(procInstId, new QueryContext(0, 1000));
-            Collection<NodeInstanceDesc> completedLogs = dataService.getProcessInstanceHistoryCompleted(procInstId, new QueryContext(0, 1000));
+            Collection<NodeInstanceDesc> activeLogs = dataService.getProcessInstanceHistoryActive(procInstId, new QueryContext(0, MAX_NODES));
+            Collection<NodeInstanceDesc> completedLogs = dataService.getProcessInstanceHistoryCompleted(procInstId, new QueryContext(0, MAX_NODES));
```

Comment:
Did you consider using paging for that?

Reply:
Can you please elaborate on this?
> yep but if the limit (X) there is for performance reason it does not make much sense to retrieve in chunks. you do it all in one go (just need to pass query context as null) .

Not sure I got it.

But yes, we have to fetch all data at once, I was mistaken by the fact that we are doing some checks for it or we are showing progress, but neither of that is true. :)

Reply:
I don't want to remove the limit (there must be a reason), but I will allow to retrieve everything if the parameter is 0 (infinity).

Reply:
Thanks, that makes sense.